client/data/settings/test/actions.js

```diff
@@ -57,30 +60,63 @@ describe( 'Settings actions tests', () => {
 		} );
 
 		test( 'before saving sets isSaving to true, and after - to false', () => {
-			apiFetch.mockReturnValue( 'api request' );
-
-			const yielded = [ ...saveSettings() ];
+			const apiResponse = {
+				data: {
+					payment_method_statuses: {
+						bancontact: 'active',
+					},
+				},
+			};
+			apiFetch.mockReturnValue( { ...apiResponse } );
 
-			const apiRequestIndex = yielded.indexOf( 'api request' );
+			const saveGenerator = saveSettings();
 
-			const isSavingStartIndex = findIndex(
-				yielded,
+			// Assert the first yield is updating isSaving to true
+			let next = saveGenerator.next();
+			expect( next.value ).toEqual(
 				updateIsSavingSettings( true, null )
 			);
 
-			const isSavingEndIndex = findIndex(
-				yielded,
+			// Execute the next step, which should be the apiFetch call
+			next = saveGenerator.next();
+			expect( next.value ).toEqual( apiResponse );
+
+			// Simulate the response from the apiFetch call and proceed to the next yield
+			// Since the actual fetching process is mocked, pass the apiResponse to the next saveGenerator step directly
+			next = saveGenerator.next( apiResponse );
+			expect( next.value ).toEqual(
+				updateSettingsValues( {
+					payment_method_statuses:
+						apiResponse.data.payment_method_statuses,
+				} )
+			);
```

Comment:
Could we use the value, instead of exporting `updateSettingsValues`?
Or, as an alternative, should we rename `updateSettingsValues` to something else (like __internal_updateSettingsValues`), to "discourage" someone from using it?

```suggestion
			expect( next.value ).toEqual( {
				type: 'SET_SETTINGS_VALUES',
				payload: {
					payment_method_statuses:
						apiResponse.data.payment_method_statuses,
				},
			} );
```

Reply:
Makes total sense to avoid exporting `updateSettingsValues`, applied!